content/casual/highlight-my-planets/plugin.js

```diff
@@ -0,0 +1,162 @@
+          value='${value}'
+          min='${min}'
+          max='${max}'
+          step='${step}'
+          style='transform: translateY(3px); margin: 0 10px;' />
+        <span>${value}</span>
+      </label>`;
+  }
+
+  getColorPicker(className, text, value) {
+    return `<label class='${className}'>
+      <div style='display: inline-block; min-width: 120px'>${text}</div><input type='color' 
+          value='${value}'
+          style='transform: translateY(3px); margin: 0 10px;' />
+      </label>`;
+  }
+
+  async render(div) {
+    div.style.width = '330px';
+    div.style.height = '100px';
+    div.style.minHeight = '100px';
+
+    const label = document.createElement('label');
+    div.innerHTML = [
+      this.getColorPicker('ownColor', 'Color:', this.ownColor),
+      this.getSliderHtml('range', 'Planet Range:', 1, 100, 1, this.rangePercent),
+      this.getSliderHtml('alpha', 'Gradient Alpha:', 0, 1, 0.01, this.alpha),
+      this.getSliderHtml('globalAlpha', 'Global Alpha:', 0, 1, 0.01, this.globalAlpha)
+    ].join('<br />');
+
+    this.valueRange = div.querySelector('label.range span');
+    this.sliderRange = div.querySelector('label.range input');
+    this.sliderRange.addEventListener('input', this.rangeHandler);
+
+    this.valueAlpha = div.querySelector('label.alpha span');
+    this.sliderAlpha = div.querySelector('label.alpha input');
+    this.sliderAlpha.addEventListener('input', this.alphaHandler);
+
+    this.valueGlobalAlpha = div.querySelector('label.globalAlpha span');
+    this.sliderGlobalAlpha = div.querySelector('label.globalAlpha input');
+    this.sliderGlobalAlpha.addEventListener('input', this.globalAlphaHandler);
+
+    this.colorPickerOwnColor = div.querySelector('label.ownColor input');
+    this.colorPickerOwnColor.addEventListener('input', this.ownColorHandler);
+  }
+
+  rangeHandler() {
+    this.rangePercent = parseInt(this.sliderRange.value);
+    this.valueRange.innerHTML = `${this.sliderRange.value}%`;
+  }
+
+  alphaHandler() {
+    this.alpha = parseFloat(this.sliderAlpha.value);
+    this.valueAlpha.innerHTML = `${this.sliderAlpha.value}`;
+  }
+
+  globalAlphaHandler() {
+    this.globalAlpha = parseFloat(this.sliderGlobalAlpha.value);
+    this.valueGlobalAlpha.innerHTML = `${this.sliderGlobalAlpha.value}`;
+  }
+
+  ownColorHandler() {
+    this.ownColor = this.colorPickerOwnColor.value;
+  }
+
+  draw(ctx) {
+    const origGlobalAlpha = ctx.globalAlpha;
+    const origFillStyle = ctx.fillStyle;
+    const origSrokeStyle = ctx.strokeStyle;
+    const viewport = ui.getViewport();
+    const planets = df.getMyPlanets();
+
+    // paint bigger ones first
+    planets.sort((a, b) => b.range - a.range);
+
+    for (const p of planets) {
+
+      // draw range circle
+      const { x, y } = viewport.worldToCanvasCoords(p.location.coords);
+      const hsl = this.hexToHsl(this.ownColor);
+
+      const fac = Math.max(0, Math.log2(this.rangePercent / 5));
```

Comment:
What do you think about replacing this line with:
`const fac = this.rangePercent*0.01;`
and setting the this.rangePercent default to 20 (in the constructor)

Reply:
Thanks for the review, you right, it's done